server/datastore/mysql/hosts.go

```diff
@@ -587,13 +592,17 @@ func (d *Datastore) GenerateHostStatusStatistics(ctx context.Context, filter fle
 		`, fleet.OnlineIntervalBuffer, fleet.OnlineIntervalBuffer, whereClause)
 
 	summary := fleet.HostSummary{TeamID: filter.TeamID}
-	err := sqlx.GetContext(ctx, d.reader, &summary, sqlStatement, now, now, now, now, now)
+	err := sqlx.GetContext(ctx, d.reader, &summary, sqlStatement, args...)
 	if err != nil && err != sql.ErrNoRows {
 		return nil, ctxerr.Wrap(ctx, err, "generating host statistics")
 	}
 
 	// get the counts per platform, the `h` alias for hosts is required so that
 	// reusing the whereClause is ok.
+	args = []interface{}{}
```

Comment:
Nit, but you can use `args = args[:0]` to reuse the existing slice.
```suggestion
	args = args[:0]
```

Reply:
Yes, but if we add another param in the future, I think this would evolve in a more clean way.